File: server/http/handler/audit.go
package handler

import (
	"net/http"
	"strconv"

	"gitee.com/openeuler/PilotGo-plugin-llmops/server/service/audit"
	"github.com/gin-gonic/gin"
)

var auditSrv = audit.GetAuditService()

func ListAuditByProjectID(c *gin.Context) {
	idStr := c.Param("id")
	id, err := strconv.Atoi(idStr)
	if err != nil || id <= 0 {
		ResponseError(c, http.StatusBadRequest, "invalid id")
		return
	}
	items, err := auditSrv.ListAuditsByProjectID(c.Request.Context(), id)
	if err != nil {
		ResponseError(c, http.StatusInternalServerError, err.Error())
		return
	}
	res := make([]gin.H, len(items))
	for i, a := range items {
		res[i] = gin.H{
			"id":          a.ID,
			"time":        a.CreatedAt,
			"operator":    a.Actor,
			"component":   a.Target,
			"description": a.Action,
		}
	}
	Response(c, res)
}

type listAuditQueryReq struct {
	ProjectID  string `form:"project_id"`
	Actor      string `form:"actor"`
	ActionType string `form:"action_type"`
	Target     string `form:"target"`
}

func ListAuditByFilters(c *gin.Context) {
	var req listAuditQueryReq
	if err := c.ShouldBindQuery(&req); err != nil {
		ResponseError(c, http.StatusBadRequest, "invalid query")
		return
	}
	var pid *int
	if req.ProjectID != "" {
		if n, err := strconv.Atoi(req.ProjectID); err == nil && n > 0 {
			pid = &n
		} else {
			ResponseError(c, http.StatusBadRequest, "invalid project_id")
			return
		}
	}
	items, err := auditSrv.ListAuditsByFilters(c.Request.Context(), &audit.ListAuditReq{
		ProjectID:  pid,
		Actor:      req.Actor,
		ActionType: req.ActionType,
		Target:     req.Target,
	})
	if err != nil {
		ResponseError(c, http.StatusInternalServerError, err.Error())
		return
	}
	Response(c, items)
}

File: web/src/apis/audit.ts
import { httpClient, type PaginationResponse } from './request'

export interface AuditItem {
    id: number | string
    time: string
    operator: string
    component: string
    description: string
}

// 后端 ListAuditByProjectID 返回 ResponsePage { page, perpage, total, data };
// data 已在 handler 映射为 time/operator/component/description,无需前端再映射。
export async function listProjectAuditLogs(
    projectId: number | string,
    page = 1,
    perpage = 20,
): Promise<PaginationResponse<AuditItem>> {
    const res = (await httpClient.get<AuditItem[]>(
        `/api/project/${projectId}/audit/logs`,
        { page, perpage },
    )) as unknown as PaginationResponse<AuditItem>
    return {
        page: res.page ?? page,
        perpage: res.perpage ?? perpage,
        total: res.total ?? 0,
        data: res.data ?? [],
    }
}

// 后端 ListAuditByFilters 打 /api/audit/logs,返回 ResponsePage { page, perpage, total, data }。
// 空串/undefined 由 httpClient.get 自动丢弃,后端 dao 亦跳过空条件,故 filters 可全传。
export async function listAuditByFilters(
    projectId: number | string,
    filters: { actor?: string; actionType?: string; target?: string },
    page = 1,
    perpage = 20,
): Promise<PaginationResponse<AuditItem>> {
    const res = (await httpClient.get<AuditItem[]>('/api/audit/logs', {
        project_id: projectId,
        actor: filters.actor,
        action_type: filters.actionType,   // camelCase 参数 → snake_case query key
        target: filters.target,
        page,
        perpage,
    })) as unknown as PaginationResponse<AuditItem>
    return {
        page: res.page ?? page,
        perpage: res.perpage ?? perpage,
        total: res.total ?? 0,
        data: res.data ?? [],
    }
}
